fix(router-core): correctly resolve custom params from declarative masks when building the masked url

hakanshehu opened the pull request:
This PR fixes #5755 , where custom params passed to declarative route masks are ignored. When the masked params differ from the original URL params, the masked URL can display `undefined` instead of the expected values.

I don't have full context on all internal router behaviors, so please treat this as a proposal. If there's a better approach, I’m happy to adjust the implementation.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Route masking supports per-mask parameter customization (merge/override/transform) and preserves search, hash, state, and original navigation info.
  * UI examples and links added to demonstrate original vs. masked pathnames.

* **Tests**
  * Added extensive unit and end-to-end tests validating mask discovery, param transformations (boolean/object/function), multiple masks, href/pathname integrity, and masking edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

schiller-manuel commented:
we definitely need tests here. ideally start with the tests first and then the solution

nlynzaad commented on packages/router-core/tests/mask.test.ts:
```diff
@@ -0,0 +1,729 @@
+    expect(location.maskedLocation!.pathname).toBe('/photos/abc123')
+    expect(location.pathname).toBe('/photos/abc123/details')
+    // Ensure no undefined values
+    expect(location.maskedLocation!.pathname).not.toContain('undefined')
```
this assertion is already covered above. if it is '/photos/abc123' then we already know it doesnt contain undefined

hakanshehu commented:
Hi @nlynzaad thanks for the feedback.

I addressed the comments in the core mask tests. I've also added some routes and a simple e2e test for react and solid. Let me know if I should use another approach or improve anything for the new tests.
